pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -650,19 +668,23 @@ func (r *ReconcilePravegaCluster) deploySegmentStore(p *pravegav1beta1.PravegaCl
 			if err != nil {
 				return err
 			}
-			if statefulSet.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort != sts.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort {
-				sts.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort = statefulSet.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort
-				sts.Spec.Template.Spec.Containers[0].ReadinessProbe = statefulSet.Spec.Template.Spec.Containers[0].ReadinessProbe
-				sts.Spec.Template.Spec.Containers[0].LivenessProbe = statefulSet.Spec.Template.Spec.Containers[0].LivenessProbe
+
+			if !r.checkVersionUpgradeTriggered(p) && !r.isRollbackTriggered(p) {
+				originalsts := sts.DeepCopy()
+				sts.Spec.Template = statefulSet.Spec.Template
 				err = r.client.Update(context.TODO(), sts)
 				if err != nil {
 					return fmt.Errorf("failed to update stateful set: %v", err)
 				}
-				err = r.restartStsPod(p)
-				if err != nil {
-					return err
+
+				if !reflect.DeepEqual(originalsts.Spec.Template, sts.Spec.Template) {
+					err = r.restartStsPod(p)
```

Comment:
`restartStsPod` will introduce a 10 minute delay if there is an issue with any pod in coming back to ready state, could we have a workaround for that?

Reply:
Raised https://github.com/pravega/pravega-operator/issues/562 to track this